ci/travis_before_script_c_glib.sh

```diff
@@ -100,7 +100,7 @@ if [ $BUILD_SYSTEM = "autotools" ]; then
 
   ./configure $CONFIGURE_OPTIONS
 
-  make -j4
+  ARCHFLAGS="-arch x86_64" make -j4
```

Comment:
It's better that we specify `ARCHFLAGS` only on macOS like the following:

```sh
if [ $TRAVIS_OS_NAME = "osx" ]; then
  ARCHFLAGS="-arch x86_64" make -j4
else
  make -j4
fi
```

Because it's needless for other platforms.

Reply:
Ah right. Thanks.